extendr-engine/src/lib.rs

```diff
@@ -1,11 +1,65 @@
-//! A singleton instance of the R interpreter.
+//! Embeds a a single R process
 //!
-//! Only call this from `main()` if you want to run stand-alone.
+//! Using R's C-API requires the embedding of the R runtime.
+//! Thus, when using bindings in `libR-sys`, it is necessary that
+//! either the a R process is the caller, or that the process instantiates
+//! an accompanying R process. Otherwise, a run-time error occurs e.g.
+//! `(signal: 11, SIGSEGV: invalid memory reference)` or
 //!
-//! Its principal use is for testing.
+//! ```text
+//! Caused by:
+//! process didn't exit successfully: `/extendr/tests/extendrtest/target/debug/deps/extendrtest-59155c3c146ae614` (signal: 11, SIGSEGV: invalid memory reference)
+//! ```
 //!
-//! See [Rembedded.c](https://github.com/wch/r-source/blob/trunk/src/unix/Rembedded.c).
+//! ## Testing
 //!
+//! Within tests, one must use [`test!`] or [`with_r`] as a wrapper around
+//! code that uses the R runtime, e.g.
+//!
+//! ```no_run
+//! #[test]
+//! fn testing_r_code() {
+//!     with_r(|| {
+//!     
+//!     });
+//! }
+//! ```
+//!
+//! Similarly with `test!` that is available in `extendr_api`, one may
+//!
+//! ```no_run   
+//! #[test]
+//! fn testing_r_code() {
+//!     test! {
+//!     
+//!     };
+//! }
+//! ```
+//!
+//! The advantage of `test!` is that it allows the use of `?` in test code, while
+//! `with_r` is not macro-based, thus code formatter `rustfmt` and rust LSPs (Rust Analyzer, Rust Rover, etc.)
+//! works within `with_r` without any problems.
```

Comment:
maybe note later that `with_r()` is therefore better.

Reply:
Pros and cons associated with both. You can't use? In withr